src/plans/components/ProviderSelect.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { ComponentProps, FC, ReactNode, useMemo } from 'react';
+import { getResourceUrl } from 'src/modules/Providers';
+
+import { ExternalLink } from '@components/common/ExternalLink';
+import Select from '@components/common/Select';
+import { EmptyState } from '@components/EmptyState';
+import { ProviderModelGroupVersionKind, ProviderModelRef, V1beta1Provider } from '@kubev2v/types';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { EmptyStateVariant, SelectOption } from '@patternfly/react-core';
+import { isEmpty } from '@utils/helpers';
+import { useForkliftTranslation } from '@utils/i18n';
+
+type ProviderSelectProps = Pick<ComponentProps<typeof Select>, 'onSelect' | 'status'> & {
+  id: string;
+  value: string;
+  namespace: string;
+  placeholder?: string;
+  emptyState?: ReactNode;
+};
+
+export const ProviderSelect: FC<ProviderSelectProps> = ({
+  id,
+  value,
+  namespace,
+  placeholder,
+  emptyState,
+  status,
+  onSelect,
+}) => {
+  const { t } = useForkliftTranslation();
+  const [providers, providersLoaded, providerError] = useK8sWatchResource<V1beta1Provider[]>({
+    namespace,
+    groupVersionKind: ProviderModelGroupVersionKind,
+    namespaced: true,
+    isList: true,
+  });
+
+  const providersListUrl = useMemo(
+    () =>
+      getResourceUrl({
+        reference: ProviderModelRef,
+        namespace: namespace,
+      }),
+    [namespace],
+  );
+
+  return (
+    <Select
+      id={id}
+      value={value}
+      status={status}
+      onSelect={onSelect}
+      placeholder={
+        placeholder ||
+        (!providersLoaded || providerError ? t('Providers failed to load.') : undefined)
+      }
+      isDisabled={!providersLoaded || providerError}
+    >
+      {isEmpty(providers) &&
+        (emptyState || (
+          <EmptyState variant={EmptyStateVariant.sm}>
+            There are no providers in project <strong>{namespace}</strong>. Create one from the{' '}
+            <ExternalLink href={providersListUrl} isInline>
+              Providers page
+            </ExternalLink>
+          </EmptyState>
+        ))}
```

Comment:
instead of branching in the return statment, this can be moved up,
`if (isEmpty(providers) && !loading) {
        return emptyState || (
          <EmptyState variant={EmptyStateVariant.sm}>
            There are no providers in project <strong>{namespace}</strong>. Create one from the{' '}
            <ExternalLink href={providersListUrl} isInline>
              Providers page
            </ExternalLink>
          </EmptyState>
        ))}
}`

Reply:
about line 46

Reply:
We are not just returning empty state when providers is empty, this is returned within the Select as a dropdown item, which is why it's nested right now.